include/z64scene.h

```diff
@@ -58,103 +58,111 @@ typedef struct {
     /* 0x04 */ Vec3s* points; // Segment Address to the array of points
 } Path; // size = 0x8
 
-// Mesh headers
+// Room shapes
 
 typedef enum {
-    /* 0 */ MESH_HEADER_TYPE_0,
-    /* 1 */ MESH_HEADER_TYPE_1,
-    /* 2 */ MESH_HEADER_TYPE_2,
-    /* 3 */ MESH_HEADER_TYPE_MAX
-} MeshHeaderType;
+    /* 0 */ ROOM_SHAPE_TYPE_NORMAL,
+    /* 1 */ ROOM_SHAPE_TYPE_IMAGE,
+    /* 2 */ ROOM_SHAPE_TYPE_CULLABLE,
+    /* 3 */ ROOM_SHAPE_TYPE_MAX
+} RoomShapeType;
 
 typedef struct {
     /* 0x00 */ u8 type;
-} MeshHeaderBase; // size = 0x01
+} RoomShapeBase; // size = 0x01
 
 typedef struct {
     /* 0x00 */ Gfx* opa;
     /* 0x04 */ Gfx* xlu;
-} MeshHeader01Entry; // size = 0x08
+} RoomShapeDListsEntry; // size = 0x08
 
 typedef struct {
-    /* 0x00 */ MeshHeaderBase base;
+    /* 0x00 */ RoomShapeBase base;
     /* 0x01 */ u8 numEntries;
-    /* 0x04 */ MeshHeader01Entry* entries;
-    /* 0x08 */ MeshHeader01Entry* entriesEnd;
-} MeshHeader0; // size = 0x0C
+    /* 0x04 */ RoomShapeDListsEntry* entries;
+    /* 0x08 */ RoomShapeDListsEntry* entriesEnd;
+} RoomShapeNormal; // size = 0x0C
 
 typedef enum {
-    /* 1 */ MESH_HEADER1_FORMAT_SINGLE = 1,
-    /* 2 */ MESH_HEADER1_FORMAT_MULTI
-} MeshHeader1Format;
+    /* 1 */ ROOM_SHAPE_IMAGE_AMOUNT_SINGLE = 1,
+    /* 2 */ ROOM_SHAPE_IMAGE_AMOUNT_MULTI
+} RoomShapeImageAmountType;
 
 typedef struct {
-    /* 0x00 */ MeshHeaderBase base;
-    /* 0x01 */ u8    format; // MeshHeader1Format
-    /* 0x04 */ MeshHeader01Entry* entry;
-} MeshHeader1Base; // size = 0x08
+    /* 0x00 */ RoomShapeBase base;
+    /* 0x01 */ u8    amountType; // RoomShapeImageAmountType
+    /* 0x04 */ RoomShapeDListsEntry* entry;
+} RoomShapeImageBase; // size = 0x08
 
 typedef struct {
-    /* 0x00 */ MeshHeader1Base base;
+    /* 0x00 */ RoomShapeImageBase base;
     /* 0x08 */ void* source;
     /* 0x0C */ u32   unk_0C;
     /* 0x10 */ void* tlut;
     /* 0x14 */ u16   width;
     /* 0x16 */ u16   height;
     /* 0x18 */ u8    fmt;
     /* 0x19 */ u8    siz;
-    /* 0x1A */ u16   mode0;
+    /* 0x1A */ u16   tlutMode;
     /* 0x1C */ u16   tlutCount;
-} MeshHeader1Single; // size = 0x20
+} RoomShapeImageSingle; // size = 0x20
 
 typedef struct {
     /* 0x00 */ u16   unk_00;
-    /* 0x02 */ u8    id;
+    /* 0x02 */ u8    bgCamIndex; // for which bg cam index is this entry for
     /* 0x04 */ void* source;
     /* 0x08 */ u32   unk_0C;
     /* 0x0C */ void* tlut;
     /* 0x10 */ u16   width;
     /* 0x12 */ u16   height;
     /* 0x14 */ u8    fmt;
     /* 0x15 */ u8    siz;
-    /* 0x16 */ u16   mode0;
+    /* 0x16 */ u16   tlutMode;
     /* 0x18 */ u16   tlutCount;
-} BgImage; // size = 0x1C
+} RoomShapeImageMultiBackgroundEntry; // size = 0x1C
```

Comment:
This name is too long imo, and we could probably shorten it with Bg and/or omitting some of the less important terms

Reply:
well every word is important, "background" or "entry" may be the least important but one of them has to stay

"RoomShape": common prefix, hard to shorten
"Image" : type of the RoomShape
"Multi" : type of the RoomShapeImage (there's also "Single")

-> `RoomShapeImageMultiBgEntry`,
could be `RoomShapeImageMultiEntry` or `RoomShapeImageMultiBg` if you really want it

Reply:
`RoomShapeImageMultiBgEntry` is probably good enough. It's still a bit on the long side but it should work

Tho I suggest applying the same change to the `Room_GetImageMultiBackgroundEntry` function (and we can change its argument to just `roomShape`, imo)